pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -289,6 +289,27 @@ public SuccessResponse updateTableConfig(
     return new SuccessResponse("Table config updated for " + tableName);
   }
 
+  @POST
+  @Path("/tables/preview")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Preview table config for a table", notes = "Preview table config for a table")
```

Comment:
Should this be named something more generic as validate as opposed to preview? Also please add more documentation, as well as description on the usage/purpose of this api.